### ISSUE TITLE
p2p: De-risk migration of the P2P layer to libp2p

### ISSUE THREAD

**Issue by thanethomson** [cometbft/cometbft#1111]
The idea of employing [libp2p](https://libp2p.io/) in the transport layer of the consensus engine has been around for quite some time (see https://github.com/tendermint/tendermint/issues/3696).

Benefits:
- We would have an alternative to the current [STS](https://en.wikipedia.org/wiki/Station-to-Station_protocol)-based mechanism, in that we could relatively easily adopt [Noise](https://noiseprotocol.org/) or TLS.
- It would potentially allow for greater modularity in the P2P layer, better enabling consensus engine developers to do new and interesting things.
- Less code for the code team to maintain, meaning more rapid value delivery elsewhere.

Drawbacks:
- Introducing new dependencies always adds risk to the project (technical and sustainability-related risk).
- Refactoring the existing P2P layer would potentially require a substantial amount of effort

Relating to the second drawback (quantifying the refactoring risk), I propose we have a spike (1 to 2 weeks max) to at least better scope the work practically. This is not meant to be a theoretical effort, but to have a working demo of the system (however hacky) running on top of libp2p by the end of the spike, along with a clear list of tasks as to how to go about productionizing the prototype so we can decide whether it's worth it.

**Comment by p-shahi** [comment]
👋 Hi @thanethomson, popping in to say hello from the go-libp2p team, great to see this spike of testing out libp2p in comet-bft! In case you would like to discuss anything related to this effort (testing, issues, etc) please feel free to reach out in the [repo](https://github.com/libp2p/go-libp2p/issues) (I'm happy to organize a sync call as well in case that'd be useful).

**Comment by cason** [comment]
Are we still taking this direction?

**Comment by sergio-mena** [comment]
@cason yes, AFAICT. We haven't taken steps in that direction due to lack of bandwidth in the team. Besides, it would be great to reuse here the experience you're gaining with libp2p in malachite.

**Comment by p-shahi** [comment]
Hey @sergio-mena @thanethomson -  I understand your team is bandwidth constrained atm but you'd still like to move in this direction - exciting! [IP Shipyard](https://ipshipyard.com/), the team maintaining go-libp2p, would love to discuss your potential migration to libp2p and see how we can help your goals.
I'd like to organize a chat between you (@sergio-mena @thanethomson, and anyone else) and the go-libp2p team. You can reach me at prithvi.shahi@ipshipyard.com. Look forward to hearing from you!

**Comment by cloud-j-luna** [comment]
Hello :wave: Has there been any updates on the spike for the introduction of libp2p into the P2P layer?
